### PR TITLE
Add an implicit resize to a rollback operation

### PR DESCRIPTION
**Pull request by ajordens**
- Will resize the currently disabled ASG such that it matches the active ASG

### CODE REVIEW

**Comment by ajordens** [comment]
Had this on my back burner for awhile ... this PR makes the explicit rollback stage a little safer by resizing the previous ASG prior to disabling the current ASG.

PTAL @cfieber @duftler @ttomsu

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/pipeline/support/ResizeStrategy.groovy [hunk 1]
```diff
@@ -52,6 +52,28 @@ interface ResizeStrategy {
     Integer min
   }
 
+  @Canonical
+  static class Source implements Serializable {
+    Collection<String> zones
+    Collection<String> regions
+    String region
+    String zone
```
Comment (ttomsu) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/pipeline/support/ResizeStrategy.groovy]: Seems like a single (or collection of) `Location` object(s) could be used here to replace all of these vars. Is there a way to tell Jackson "anytime you see `region` give me a `Location.region(region)`?"

**Comment by ttomsu** [comment]
Outside of comment, LGTM.

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/pipeline/support/ResizeStrategy.groovy (hunk 1 above)
Comment (ajordens) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/pipeline/support/ResizeStrategy.groovy]: Ya, this is unfortunate.

I don't think there's a way we can do it explicitly within Jackson (maybe @cfieber knows) ... I've not looked into custom deserializers or anything.

What could be done is a transformation when the pipeline/task first starts that say injects `_location` or `_locations` into the JSON if it ever sees zones/regions/region/zone. 

I say `_location` just to have a key that isn't going to collide with a normal user-provided value.

**Comment by ajordens** [comment]
I would also like to make our ResizeAsg operation in clouddriver not require asgName.

**Pull request merged**